genesis/utils/array_class.py

```diff
@@ -20,6 +20,10 @@
 DATA_ORIENTED = partial(dataclasses.dataclass, frozen=True) if gs.use_ndarray else ti.data_oriented
 
 
+def _s(shape, is_on):
+    return shape if is_on else (1,)
+
+
```

Comment:
I'm not in favour of using cryptic helpers. It just makes everything harder to read.